app/uk/gov/hmrc/gform/sharedmodel/graph/DependencyGraph.scala

```diff
@@ -185,7 +185,28 @@ object DependencyGraph {
         .flatten
 
     val emptyGraph: Graph[GraphNode, DiEdge] = Graph.empty
-    emptyGraph ++ (allEdges ++ addToListEdges)
+    val graph = emptyGraph ++ (allEdges ++ addToListEdges)
+
+    def govukListEdges(g: GraphNode, fcId: FormComponentId): List[DiEdge[GraphNode]] =
+      g ~> GraphNode.Expr(FormCtx(fcId)) ::
+        GraphNode.Expr(FormCtx(fcId)) ~> GraphNode.Simple(fcId) ::
+        formModel.allFormComponents
+          .filter(_.baseComponentId === fcId.baseComponentId)
+          .flatMap(fc =>
+            List(
+              GraphNode.Simple(fcId) ~> GraphNode.Expr(FormCtx(fc.id)),
+              GraphNode.Expr(FormCtx(fc.id)) ~> GraphNode.Simple(fc.id)
+            )
+          )
+    val govukListNodes = graph.nodes.flatMap { n =>
+      n.value match {
+        case g @ GraphNode.Expr(NumberedList(fcId)) => govukListEdges(g, fcId)
+        case g @ GraphNode.Expr(BulletedList(fcId)) => govukListEdges(g, fcId)
+        case e                                      => List.empty[DiEdge[GraphNode]]
+      }
+    }.toSet
```

Comment:
Why is this here?

Reply:
alternatively we can evaluate `FormCtx(addressid)` to AddressResult(addresslines)` or `ListResult(List(AddressResult))` but we will need to review and update all the usages as they all expect `AddressResult(addresslines)`

Reply:
> For `addressId` that refers to ATL address we need `ListResult(AddressResult(...))`

And this is exactly what we need to fix. [This should return]( https://github.com/hmrc/gform-frontend/blob/46f798b67926b82957b1e6ec4aefe5ecf535429a/app/uk/gov/hmrc/gform/eval/EvaluationResults.scala#L355) `ListResult(AddressResult(...))` if `addressId` is ATL component.